### PR TITLE
feat(backend): serve keys from payment pointer url

### PR DESCRIPTION
**Pull request by omertoast**
## Changes proposed in this pull request

- add `paymentPointerUrl` field to `clients` table. This is temporary as I don't want to break services that use clients, if you have a better way to link clients to payment pointers I'm open to it.
- add `getClientByPaymentPointerUrl` method to client service.
- add `getPaymentPointerKeys` method to `paymentPointerRoutes` to serve key set.
- add `/jwks.json` route.

## Context

This is one of the first tasks for refactoring the Key Directory from v1 to v2. As the first step, I'm implementing `/jkws.json` relative path to the payment pointer URL to serve the client key set linked to the payment pointer.
 
## TODO

- [x] Link clients and payment pointers
- [x] Allow getting clients using payment pointers
- [x] Extract JWKS from the key set
- [x] Serve key set from `/jwks.json` endpoint.

## Checklist

- [ ] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

**Comment by adrianhopebailie** [comment]
LGTM but I suggest either @mkurapov , @sabineschaller or @wilsonianb have a quick glance over

**Review by wilsonianb — COMMENTED** [review]
Any reason not to completely replace the existing clients service?
- remove `/keys`
- just add/manage keys on the payment pointer
- store whatever client display info on the payment pointer

Or does there need to be some distinct state of a payment pointer being a client or not?

File: packages/backend/migrations/20220818141305_create_clients_table.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ exports.up = function (knex) {
     table.string('email').notNullable()
     table.string('image').notNullable()
 
+    table.string('paymentPointerUrl').notNullable()
```
[the remark this answers is not in the document]
Comment (omertoast) [reply packages/backend/migrations/20220818141305_create_clients_table.js:9]: I didn't want to remove client service in a single PR as I afraid of it's being a massive PR and breaking the other PRs. But if everyone is okay with replacing clients service with payment pointer I can go for it.

**Review by mkurapov — COMMENTED** [review]
I was able to create a client, add key to client, and get the keys via `/jwks.json` path successfully. For this PR, just should get Brandon's suggested change in there and good to go

**Comment by omertoast** [comment]
> I was able to create a client, add key to client, and get the keys via `/jwks.json` path successfully. For this PR, just should get Brandon's suggested change in there and good to go

I started a branch for moving `clients` functions and table into the `paymentPointer`. Should be ready end of the day.

**Review by mkurapov — APPROVED** [review]

**Pull request closed without merging**